src/Database/Dialect/SqlserverDialectTrait.php

```diff
@@ -102,7 +104,31 @@ public function _version()
     protected function _pagingSubquery($original, $limit, $offset)
     {
         $field = '_cake_paging_._cake_page_rownum_';
-        $order = $original->clause('order') ?: new OrderByExpression('(SELECT NULL)');
+
+        if ($original->clause('order')) {
+            // SQL server does not support column aliases in OVER clauses.  But
+            // the only practical way to specify the use of calculated columns
+            // is with their alias.  So substitute the select SQL in place of
+            // any column aliases for those entries in the order clause.
+            $select = $original->clause('select');
+            $order = new OrderByExpression();
+            $original
+                ->clause('order')
+                ->iterateParts(function ($direction, $orderBy) use ($select, $order) {
+                    $key = $orderBy;
+                    if (isset($select[$orderBy]) &&
+                        $select[$orderBy] instanceof ExpressionInterface
+                    ) {
+                        $key = $select[$orderBy]->sql(new ValueBinder());
```

Comment:
@markstory - if you can give me an example of a select statement using a calculated value and bound parameters I'd be happy to see if I can fix it for that case.  I'm still struggling to understand all the details of how to build SQL via the (new to me) V3 ORM stuff.  There are so may variations in syntax and approach that it can be hard to sort out the details.